policies/design-process.md

```diff
@@ -0,0 +1,154 @@
+Policy for Creating Designs
+===========================
+
+From time to time the OTC may require that certain problems should have a
+design document created for them before implementation should start. In that
+case this policy will apply.
+
+Pull Requests that implement a solution, or part of a solution, covered by a
+design document should clearly reference that design document. Pull requests
+that modify a solution covered by a design document should not be accepted
+unless the design document is also updated to describe the modified solution.
+
+The design approach described in this document should be considered iterative.
+Only "just enough" design should be done to get the implementation project to
+the next stage. Certain design decisions and sub-problems may be deferred until
+some later stage.
+
+Inputs
+------
+
+Prior to commencing design work the OTC shall identify:
+
+1) The business requirements for the problem to be solved (obtained from the OMC)
+2) The scope of the specific problem area to be addressed
+3) Any additional design constraints that the OTC wishes to impose
+4) Any other constraints that may apply (e.g. timescales)
+5) The individual(s) that will work on developing the design (referred to as the
+   "design team" in this document)
+
+Typically these will be identified during an OTC meeting and documented in the
+meeting minutes.
+
+Outputs
+-------
+
+The result of this process should be a design document in markdown format,
+reviewed via a pull request and approved for merge into the main source git
+repository. The design document must include:
+
+ - Any assumptions that have been made
+ - Any omissions that may be filled in later
+ - Any APIs that the implementation will provide
```

Comment:
Does this mean that (minimal) documentation should written for the API as part of the design document?

Reply:
I've added this sentence to the document to clarify this "This should include some minimal documentation of the APIs (not necessarily end-user focussed) to show how they should be used."